devlogs/raisa/daily-logs/2025-03-16.md

```
@@ -0,0 +1,320 @@
+“**Derived or child classes must be substitutable for their base or parent classes**“
+
+Breaking it down simply:
+
+- If **class B** is a subclass of **class A**, then we should be able to replace **A** with **B** *without breaking the functionality*.
+- The **child class** should **extend** the parent class **without changing its expected behavior**.
+
+**An Example of Violation**
+
+If you extend the above rectangle class to a square class and make it have equale sides, it will violate the characteristics of the rectangle class. If you need a square class, create a seperate one.
+
+### **How to Follow LSP?**
```

markdownlint-cli2 (0.17.2)

Heading levels should only increment by one level at a time
Expected: h2; Actual: h3

(MD001, heading-increment)